kv/replica_slice.go

```diff
@@ -0,0 +1,123 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Tobias Schottdorf (tobias.schottdorf@gmail.com)
+// Author: Peter Mattis (peter@cockroachlabs.com)
+
+package kv
+
+import (
+	"github.com/cockroachdb/cockroach/gossip"
+	"github.com/cockroachdb/cockroach/proto"
+	"github.com/cockroachdb/cockroach/util/log"
+)
+
+// replicaInfo extends the Replica structure with the associated node
+// descriptor.
+type replicaInfo struct {
+	proto.Replica
+	NodeDesc proto.NodeDescriptor
+}
+
+func (i *replicaInfo) attrs() []string {
+	return i.NodeDesc.GetAttrs().Attrs
```

Comment:
Yes, that was my thought: this is only allowing matching against node attrs. Pretty sure that was the previous behavior as well.